go/datastore/internal/controller/postgresuser_controller.go

```diff
@@ -158,31 +160,54 @@ func (r *PostgresUserReconciler) Reconcile(ctx context.Context, req ctrl.Request
 	return jitterRequeue(requeueDefault), nil
 }
 
-func (r *PostgresUserReconciler) handleDelete(ctx context.Context, user *v1alpha1.PostgresUser) error {
+func (r *PostgresUserReconciler) handleDelete(ctx context.Context, user *v1alpha1.PostgresUser) (*ctrl.Result, error) {
 	credentials := new(v1alpha1.PostgresCredentials)
 	err := r.Get(ctx, types.NamespacedName{Name: user.Spec.CredentialsRef.Name, Namespace: user.Namespace}, credentials)
 
 	if err != nil || !meta.IsStatusConditionTrue(credentials.Status.Conditions, v1alpha1.ReadyConditionType.String()) {
 		if err := deleteRefSecret(ctx, r.Client, user.Namespace, user.Spec.PasswordSecretKeyRef.Name, PostgresSecretProtectionFinalizerName); err != nil {
-			return err
+			return nil, err
 		}
 		controllerutil.RemoveFinalizer(user, PostgresUserProtectionFinalizerName)
-		return nil
+		return nil, nil
+	}
+
+	if len(user.Spec.Databases) > 0 {
+		databaseList := containers.ToSet[string](user.Spec.Databases)
+
+		dbList := &v1alpha1.PostgresDatabaseList{}
+		if err := r.List(ctx, dbList, client.InNamespace(user.Namespace)); err != nil {
+			return nil, err
+		}
+		var deletingAny bool
+		for _, db := range dbList.Items {
+			if databaseList.Has(db.DatabaseName()) {
+				deletingAny = true
+				if db.DeletionTimestamp.IsZero() {
```

Comment:
I don't think we should manually delete the database in this controller, is it necessary?